xarray/core/variable.py

```diff
@@ -351,10 +300,13 @@ def _as_array_or_item(data):
     """
     data = np.asarray(data)
     if data.ndim == 0:
-        if data.dtype.kind == "M":
-            data = np.datetime64(data, "ns")
-        elif data.dtype.kind == "m":
-            data = np.timedelta64(data, "ns")
+        kind = data.dtype.kind
+        if kind in "mM":
+            unit = np.datetime_data(data.dtype)[0]
```

Comment:
For readability, prefer using tuple unpacking:
```suggestion
            unit, _ = np.datetime_data(data.dtype)
```